### ISSUE TITLE
Arrays as attributes

### ISSUE THREAD

**Issue by matthew-d-jones** [ess-dmsc/kafka-to-nexus#74]
Attributes must be allowed to be arrays, for example for the `vector` attributes of an `NXtransformation`:

```json
{
  "values": 12.064,
  "type": "dataset",
  "name": "location",
  "dataset": {
    "type": "double"
  },
  "attributes": {
    "transformation_type": "translation",
    "units": "m",
    "NX_class": "NXtransformation",
    "depends_on": ".",
    "vector": [
      0.0,
      0.0,
      -1.0
    ]
  }
}
```
Blocked by #56

**Comment by matthew-d-jones** [comment]
I'm failing to write any arrays, will create a new ticket when I've figured out what is going on.

**Comment by matthew-d-jones** [comment]
This is required to support writing the example ESS NeXus files, so I'm marking as high priority.

**Comment by dominikwerder** [comment]
👍

**Comment by matthew-d-jones** [comment]
Still to do:
- documentation
- more refactoring?
